artifact/v3/header/header_test.cpp

```diff
@@ -0,0 +1,368 @@
+	EXPECT_THAT(
+		header.artifactScripts.value().at(0),
+		testing::AnyOf(
+			testing::EndsWith("ArtifactInstall_Enter_01_test-dummy"),
+			testing::EndsWith("ArtifactInstall_Enter_02_test-dummy")));
+	EXPECT_THAT(
+		header.artifactScripts.value().at(1),
+		testing::AnyOf(
+			testing::EndsWith("ArtifactInstall_Enter_01_test-dummy"),
+			testing::EndsWith("ArtifactInstall_Enter_02_test-dummy")));
+
+	//
+	// Sub-headers
+	//
+	EXPECT_EQ(header.subHeaders.size(), 1);
+
+	//
+	// type-info
+	//
+
+	EXPECT_EQ(header.subHeaders.at(0).type_info.type, "dummy-update-module");
+
+	// type_info/0000/artifact_provides
+	EXPECT_TRUE(header.subHeaders.at(0).type_info.artifact_provides);
+	EXPECT_EQ(
+		header.subHeaders.at(0)
+			.type_info.artifact_provides.value()["rootfs-image.dummy-update-module.version"],
+		"header-tester-name");
+
+	// type_info/0000/artifact_depends
+	EXPECT_TRUE(header.subHeaders.at(0).type_info.artifact_depends);
+	EXPECT_EQ(header.subHeaders.at(0).type_info.artifact_depends.value()["foo"], "bar");
+
+	// type_info/0000/clears_artifact_provides
+	EXPECT_TRUE(header.subHeaders.at(0).type_info.clears_artifact_provides);
+	EXPECT_EQ(
+		header.subHeaders.at(0).type_info.clears_artifact_provides.value().at(0),
+		"rootfs-image.dummy-update-module.*");
+
+	// meta-data
+}
+
+
+TEST_F(HeaderTestEnv, TestTwoArtifactScriptsSuccess) {
+	mendertesting::TemporaryDirectory tmpdir {};
+	CreateTestArtifact(
+		tmpdir,
+		"rootfs-image",
+		{
+			{R"(--script ${DIRNAME}/ArtifactInstall_Enter_01_test-dummy)"},
+			{R"(--script ${DIRNAME}/ArtifactInstall_Enter_02_test-dummy)"},
+		});
+
+
+	std::fstream fs {tmpdir.Path() + "/header.tar"};
+
+	mender::common::io::StreamReader sr {fs};
+
+	ExpectedHeader expected_header = header::Parse(sr);
+
+	ASSERT_TRUE(expected_header) << expected_header.error().message;
+
+	auto header = expected_header.value();
+
+	ASSERT_TRUE(header.artifactScripts);
+	EXPECT_EQ(header.artifactScripts.value().size(), 2);
+}
+
+TEST_F(HeaderTestEnv, TestOneArtifactScripts) {
+	mendertesting::TemporaryDirectory tmpdir {};
+	CreateTestArtifact(
+		tmpdir,
+		"rootfs-image",
+		{
+			{R"(--script ${DIRNAME}/ArtifactInstall_Enter_01_test-dummy)"},
+		});
+
+	std::fstream fs {tmpdir.Path() + "/header.tar"};
+
+	mender::common::io::StreamReader sr {fs};
+
+	ExpectedHeader expected_header = header::Parse(sr);
+
+	ASSERT_TRUE(expected_header) << expected_header.error().message;
+
+	auto header = expected_header.value();
+
+	ASSERT_TRUE(header.artifactScripts);
+	EXPECT_EQ(header.artifactScripts.value().size(), 1);
+}
+
+TEST_F(HeaderTestEnv, TestHeaderNoExtraData) {
+	mendertesting::TemporaryDirectory tmpdir {};
+	CreateTestArtifact(tmpdir, "module-image", {"--type test-module-image"});
+
+	std::fstream fs {tmpdir.Path() + "/header.tar"};
+
+	mender::common::io::StreamReader sr {fs};
+
+	ExpectedHeader expected_header = header::Parse(sr);
+
+	ASSERT_TRUE(expected_header) << expected_header.error().message;
+}
```

Comment:
Perhaps add test cases for:

* Some files out of order.
* Wrong index of header or payload.